src/main/java/com/smoothstack/utopiaairlines/controller/AdminController.java

```diff
@@ -0,0 +1,179 @@
+	@PutMapping("/routes/airplane/update/id/capacity/{id}/{capacity}")
+	public String updateAirplaneById(@PathVariable Integer id, @PathVariable Integer capacity) {
+		airplaneService.updateAirplaneById(id, capacity);
+		return "Airplane Successfully Upated!";
+	}
+
+
+
+	// Delete Airplane
+	@DeleteMapping("/routes/airplane/delete/{id}")
+	public String deleteAirplane(@PathVariable Integer id) {
+		airplaneService.deleteAirplane(id);
+		return "Airplane Deleted!";
+	}
+
+	//-------------------------------------------------------------------------------------------------------------------------------------
+	// CRUD Flight
+
+
+	// Add
+	@PostMapping("/routes/flight/add")
+	public String saveFlight(@RequestBody Flight flight) {
+		flightService.addFlight(flight);
+		return "Flight Successfully Added!";
+	}
+
+	// Read All Flights
+	@GetMapping("/routes/flight/read")
+	public List<Flight> getFlights() {
+		List<Flight> flights = new ArrayList<>();
+		flights = flightService.findFlights();
+		return flights;
+	}
+
+
+	// Read Flights by Route ID 
+	@GetMapping("/routes/flight/read/route_id/{routeId}")
+	public List<Flight> getFlightsByRouteId(@PathVariable Integer routeId) {
+		List<Flight> flights = new ArrayList<>();
+		flights = flightService.findFlightsByRouteId(routeId);
+		return flights;
+	}
+
+
+	// Read Flights by Airplane ID
+	@GetMapping("/routes/flight/read/airplane_id/{airplaneId}")
+	public List<Flight> getFlightsByAirplaneId(@PathVariable Integer airplaneId) {
+		List<Flight> flights = new ArrayList<>();
+		flights = flightService.findFlightsByAirplaneId(airplaneId);
+		return flights;
+	}
+	
+	// Read Flights by Date Time 
+	@GetMapping("/routes/flight/read/date_time/{dateTime}")
+	public List<Flight> getFlightsByDateTime(@PathVariable @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME) LocalDateTime dateTime) {
+		List<Flight> flights = new ArrayList<>();
+		flights = flightService.findFlightsByDateTime(dateTime);
+		return flights;
+	}
+
+
+	//Update Whole Flight by Flight ID
+	@PutMapping("/routes/flight/update")
+	public String updateFlight(@RequestBody Flight flight) {
+		flightService.updateFlight(flight);
+		return "Flight Successfully Upated!";
+	}
+	
+	//Update Route ID by Flight ID
+	@PutMapping("/routes/flight/update/flight_id/route_id/{id}/{routeId}")
+	public String updateRouteIdByFlightId(@PathVariable Integer id, @PathVariable Integer routeId) {
+		flightService.updateRouteIdByFlightId(id, routeId);
+		return "Route ID Successfully Upated!";
+	}
+	
+	//Update Airplane ID by Flight ID
+	@PutMapping("/routes/flight/update/flight_id/airplane_id/{id}/{airplaneId}")
+	public String updateAirplaneIdByFlightId(@PathVariable Integer id, @PathVariable Integer airplaneId) {
+		flightService.updateAirplaneIdByFlightId(id, airplaneId);
+		return "Airplane ID Successfully Upated!";
+	}
+	
+	
+	//Update Departure Date Time by Flight ID
+	@PutMapping("/routes/flight/update/flight_id/departure_time/{id}/{dateTime}")
```

Comment:
I won't make a fuss about this, but for the capstone project, consider using query parameters

Reply:
Done